src/aria2p/interface.py

```diff
@@ -577,6 +629,65 @@ def process_keyboard_event_main(self, event):
         elif event.key_code in Keys.UN_SELECT_ALL:
             pass  # TODO
 
+        elif event.key_code in Keys.MOVE_HOME:
+            if self.focused > 0:
+                self.focused = self.rows.index(self.rows[0])
+                logger.debug(f"Move focus home: {self.focused}")
+                if self.focused < self.row_offset:
+                    self.row_offset = self.focused
+                elif self.focused >= self.row_offset + (self.height - 1):
+                    self.row_offset = self.focused + 1 - (self.height - 1)
+                self.follow = None
+                self.refresh = True
+
+        elif event.key_code in Keys.MOVE_END:
+            if self.focused < len(self.rows) - 1:
+                self.focused = self.rows.index(self.rows[-1])
+                logger.debug(f"Move focus end: {self.focused}")
+                if self.focused - self.row_offset >= (self.height - 1):
+                    self.row_offset = self.focused + 1 - (self.height - 1)
+                self.follow = None
+                self.refresh = True
+
+        elif event.key_code in Keys.MOVE_UP_STEP:
+            if self.focused > 0:
+                self.focused -= len(self.rows) // 5
+                if self.focused < 0:
+                    self.focused = self.rows.index(self.rows[0])
```

Comment:
Again, unless I missed something, the index of the 0th row is 0.

```suggestion
                    self.focused = 0
```